mypy/messages.py

```diff
@@ -203,18 +203,19 @@ def has_no_attr(self, original_type: Type, typ: Type, member: str, context: Cont
                     format_type(original_type)), context)
             else:
                 self.fail('Value of type {} is not indexable'.format(
-                    format_type(original_type)), context)
+                    format_type(original_type)), context, code=codes.INDEX)
         elif member == '__setitem__':
             # Indexed set.
-            self.fail('Unsupported target for indexed assignment', context)
+            self.fail('Unsupported target for indexed assignment', context, code=codes.INDEX)
         elif member == '__call__':
             if isinstance(original_type, Instance) and \
                     (original_type.type.fullname() == 'builtins.function'):
                 # "'function' not callable" is a confusing error message.
                 # Explain that the problem is that the type of the function is not known.
-                self.fail('Cannot call function of unknown type', context)
+                self.fail('Cannot call function of unknown type', context, code=codes.OPERATOR)
             else:
-                self.fail('{} not callable'.format(format_type(original_type)), context)
+                self.fail('{} not callable'.format(format_type(original_type)), context,
+                          code=codes.OPERATOR)
```

Comment:
This make me think about how we will evolve the error codes. Imagine we will decide next year that `codes.OPERATOR` is too general, but some people have a hundred of `type: ignore[operator]`. If we will introduce two new error codes instead, the existing ignores will stop working.

Do we have a mechanism to allow this? For example sub-codes like `operator:arithmetic` or allowing aliases (to rename codes).

Reply:
I don't expect that we'll need to split the error codes much further. I have some ideas but I'd want to make most of the changes before the next release. If we rename error codes, it will be easy enough to add support for aliases in the future.

Reply:
OK, we would also need to add docs before the next release.